content/BPIP-4.md

```diff
@@ -0,0 +1,93 @@
+---
+bpip: 4
+title:  Support price discovery 
+discussions-to: 
+status: Living
+created: 2023-01-25
+---
+
+## Abstract
+
+This proposal describe necessary changes on protocol in order to support price discovery. 
+
+## Motivation
+
+Currently protocol doens't supports price discovery because price is a immutable value set on offer creation.
+
+## Specification 
+
+### Options:
+
+1. **Turn Boson Vouchers compatible.**<br>
+    Boson Vouchers are ERC721 and therefore it's easier to make them compatible with price discovery features such as AMMs and Auctions mechanism. 
+    NFT pools work just ERC20 tools. Pools that are willing to buy or sell NFTs will return the same price no matter which NFT is sent in or out from the collection. 
+    That makes the current implementation of Boson Vouchers incompatible with NFT pools because Vouchers can represent different offers with different initial prices.
+    To make it compatible with AMMs, we must turn Boson Vouchers collections individual to offers - instead of sellers, so the initial price for the entire vouchers in the pool is the same (the offer price). 
+
+    **How do we communicate the price paid for the voucher to the protocol?**
+    The price is a crutial info for the protocol as we still want that protocol guarantees remain unbroken, so when a sale happens outside the protocol (no matter how this sale happens, can be seller selling preminted
+    vouchers on marketplaces, an auction, an AMM pool, a secundary sale using a settlement protocol, etc) we need inform protocol about the exchange price.
+
+    In order to make this happens we would have to request both parties (buyer and seller) to confirm the exchange. 
+    Just as proposed for [sequential commit](https://docs.google.com/document/d/1lb6ERrGtOg2iPjf17CfHJRuKWwPdkvQRsLQ3yV_0nh0/edit#heading=h.e4aw2z4amzsr),
```

Comment:
Sequential commit document is not public, so it does not make sense to reference it here.

Reply:
deleted